autoinstrumentation/java/Dockerfile

```diff
@@ -4,4 +4,4 @@ ARG version
 
 ADD https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v$version/opentelemetry-javaagent.jar /javaagent.jar
 
-RUN chmod -R o+r /javaagent.jar
+RUN chmod -R go+r /javaagent.jar
```

Comment:
Could you please explain why `o`/`others` is not sufficient? I think it should be sufficient. 

> Any other user who has access to a file. This person has neither created the file, nor he belongs to a usergroup who could own the file. Practically, it means everybody else. Hence, when you set the permission for others, it is also referred as set permissions for the world.

Reply:
from https://www.guru99.com/file-permissions.html

Reply:
thanks for the explanation, now I would like to do 2 things:
1. add the same to other instrumentations
2. test this in e2e tests - most likely add security context here https://github.com/open-telemetry/opentelemetry-operator/blob/main/tests/e2e/instrumentation-java/01-install-app.yaml

Reply:
Here are docs for the security context https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod

> In the configuration file, the runAsUser field specifies that for any Containers in the Pod, all processes run with user ID 1000. The runAsGroup field specifies the primary group ID of 3000 for all processes within any containers of the Pod. If this field is omitted, the primary group ID of the containers will be root(0). Any files created will also be owned by user 1000 and group 3000 when runAsGroup is specified. Since fsGroup field is specified, all processes of the container are also part of the supplementary group ID 2000. The owner for volume /data/demo and any files created in that volume will be Group ID 2000.

My understanding is that when the `fsGroup` is set then copied `javaagent.jar` file will inherit that group. But I still don't understand why the file cannot be read if the file has `o+r` set? cc) @jpkrohling

Reply:
Because user belongs to fsGroup and "others" means everybody else. That's what I see happening at least without group read permissions. 
I'll try to add an e2e test, but not sure I can finish that before the holidays, so it might have to wait until January. 
Thanks for getting back to me so quickly.

Reply:
`o` means "everyone else" (any _other_ user), not "everyone". You can try that locally, like this (if you don't have the apache user/group locally, use some other group that your account doesn't belong):

```
$ echo contents > testfile 
$ ls -lah testfile
-rw-r--r--. 1 jpkroehling wheel 9 Dec 17 11:18 testfile
$ sudo chown apache:apache testfile
$ sudo chmod 604 testfile
$ ls -lah testfile
-rw----r--. 1 apache apache 9 Dec 17 11:18 testfile
$ cat testfile
contents
$ sudo chown apache:wheel testfile
$ groups
wheel libvirt
$ cat testfile
cat: testfile: Permission denied
```

Reply:
I think I get it now, in the last example the `jpkroehling` is part of the `wheel` group but the file does not give permission to the group. If `jpkroehling` was removed from the `wheel` group it could read the file. 

thanks!